src/streaming/zarrv3.array.writer.hh

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "array.writer.hh"
+
+namespace zarr {
+struct ZarrV3ArrayWriter final : public ArrayWriter
+{
+  public:
+    ZarrV3ArrayWriter(ArrayWriterConfig&& config,
```

Comment:
`ArrayWriterConfig` has a `std::unique_ptr<ArrayDimensions>` member, though the next PR reveals this does need to be a shared pointer after all, because the Stream object needs to hold on to them as well for metadata purposes.